### PR TITLE
Update build.yaml

### LINKED ISSUE #109: Add windows to the CI Matrix
Windows was quickly removed from the CI matrix, because tests failed early on, and I didn't have time to investigate at the time. But we should re-enable them in https://github.com/redhat-developer/vscode-granite/blob/main/.github/workflows/build.yaml#L21, and ensure tests now work

### PR DESCRIPTION
**Pull request by athirakm94**
Add windows to the CI Matrix

Fixes #109

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,24 +1,25 @@
-# This is a basic workflow
+# Combined CI workflow for Windows, macOS, and Linux
 name: CI
 
-# Triggers the workflow on all pushes or pull requests on the main branch
+# Triggers the workflow on all pushes or pull requests on specified branches
+# Triggers the workflow on all pushes or pull requests on any branch
```
Comment (fbricon) [inline .github/workflows/build.yaml:5]: there's one superfluous comment (specified branches)

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -40,34 +41,33 @@ jobs:
 
       # Generate .vsix package for the extension 
       - name: Generate .vsix package
-        if: runner.os == 'Linux'
```
Comment (fbricon) [inline .github/workflows/build.yaml:43]: no there's no need to generate the vsix 3 times. So only linux generates and uploads the archive

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -40,34 +41,33 @@ jobs:
...
       - name: Generate .vsix package
-        if: runner.os == 'Linux'  
         run: |
           npm install -g @vscode/vsce
           vsce package
 
-      # Upload VSIX artifact 
+      # Upload VSIX artifact
       - name: Upload VSIX artifact
-        if: runner.os == 'Linux'
```
Comment (fbricon) [inline .github/workflows/build.yaml:50]: no, see previous comment

**Review by fbricon — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 3 above)
Comment (athirakm94) [reply .github/workflows/build.yaml:50]: Linux support has been re-added for generating and uploading artifact

**Review by fbricon — APPROVED** [review]

**Comment by fbricon** [comment]
thanks @athirakm94

**Pull request merged**